### PR TITLE
Add recommended k8s labels

### PR DESCRIPTION
**Pull request by eddycharly**
This PR adds recommended k8s labels on resources.
Implementation for proposal https://github.com/tektoncd/pipeline/issues/2497

Related PR in pipelines: https://github.com/tektoncd/pipeline/pull/2501

Base: added labels and updated selectors for all resources in `/config`

NOTE: this will introduce a breaking change when trying to update deployments as label selectors have changed.

### CODE REVIEW

File: config/101-podsecuritypolicy.yaml [hunk 1]
```diff
@@ -16,6 +16,11 @@ apiVersion: policy/v1beta1
 kind: PodSecurityPolicy
 metadata:
   name: tekton-triggers
+  labels:
+    app.kubernetes.io/name: psp
```
Comment (vdemeester) [inline config/101-podsecuritypolicy.yaml:20]: Any reason for using `psp` instead of a "long" name ?

Comment (eddycharly) [reply config/101-podsecuritypolicy.yaml:20]: Shall i add the `app.kubernetes.io/name` label at all ?

Comment (eddycharly) [reply config/101-podsecuritypolicy.yaml:20]: I mean, we want to use label for two things:
- identify the application/component the resource belongs to
- be able to look up the resource when necessary

I don’t think we need to look up PSP anywhere and other labels are enough to identify the application so...

Comment (vdemeester) [reply config/101-podsecuritypolicy.yaml:20]: yeah, I don't think `app.kubernetes.io/name` has too much value indeed.

**Pull request merged**